docs/using-the-sdk/cosmovisor.md

```diff
@@ -0,0 +1,174 @@
+│   └── bin
+│       └── $DAEMON_NAME
+└── upgrades
+    └── <name>
+        └── bin
+            └── $DAEMON_NAME
+```
+
+Each version of the Cosmos SDK application is stored under either `genesis` or `upgrades/<name>`, which holds `bin/$DAEMON_NAME`
+along with any other needed files such as auxiliary client programs or libraries. `current` is a symbolic link to the currently
+active folder (so `current/bin/$DAEMON_NAME` is the currently active binary).
+
+*Note: the `name` variable in `upgrades/<name>` holds the URI-encoded name of the upgrade as specified in the upgrade module plan.*
+
+Please note that `$DAEMON_HOME/cosmovisor` just stores the *binaries* and associated *program code*.
+The `cosmovisor` binary can be stored in any typical location (eg `/usr/local/bin`). The actual blockchain
+program will store it's data under their default data directory (e.g. `$HOME/.gaiad`) which is independent of
+the `$DAEMON_HOME`. You can choose to set `$DAEMON_HOME` to the actual binary's home directory and then end up
+with a configuation like the following, but this is left as a choice to the system admininstrator for best
+directory layout:
+
+```
+.gaiad
+├── config
+├── data
+└── cosmovisor
+```
+
+## Usage
+
+The system administrator admin is responsible for:
+* installing the `cosmovisor` binary and configure the host's init system (e.g. `systemd`, `launchd`, etc) along with the environmental variables appropriately;
+* installing the `genesis` folder manually;
+* installing the `upgrades/<name>` folders manually.
+
+`cosmovisor` will set the `current` link to point to `genesis` at first start (when no `current` link exists) and handles
+binaries switch overs at the correct points in time, so that the system administrator can prepare days in advance and relax at upgrade time.
+
+Note that blockchain applications that wish to support upgrades may package up a genesis `cosmovisor` tarball with this information,
+just as they prepare the genesis binary tarball. In fact, they may offer a tarball will all upgrades up to current point for easy download
+for those who wish to sync a fullnode from start.
+
+The `DAEMON` specific code and operations (e.g. tendermint config, the application db, syncing blocks, etc) are performed as normal.
+Application binaries' directives such as command-line flags and environment variables work normally.
+
+## Example: simd
+
+The following instructions provide a demonstration of `cosmovisor`'s integration with the `simd` application
+shipped along the Cosmos SDK's source code.
+
+First compile `simd`:
+
+```
+cd cosmos-sdk/
+make build
+```
+
+Create a new key and setup the `simd` node:
+
+```
+rm -rf $HOME/.simapp
+./build/simd keys --keyring-backend=test add validator
+./build/simd init testing --chain-id test
+./build/simd add-genesis-account --keyring-backend=test $(./build/simd keys --keyring-backend=test show validator -a) 1000000000stake,1000000000validatortoken
+./build/simd gentx validator --chain-id test
```

Comment:
nit: This needs the `--keyring-backend test` flag to be consistent with the `add-genesis-account` command

Reply:
fixed, please check again

Reply:
ace :bow: